app/models/calculation_result.rb

```diff
@@ -280,32 +281,46 @@ def capital_rows(prefix:)
     capital_row_items(prefix:).transform_values { |value| monetise(value) }
   end
 
-  def main_home_rows(prefix:)
-    main_home = capital_items(:properties, prefix).fetch(:main_home)
-    disregard = main_home.fetch(:net_equity) - main_home.fetch(:assessed_equity)
-    data = {
-      value: monetise(main_home.fetch(:value)),
-      mortgage: monetise(-main_home.fetch(:outstanding_mortgage)),
-      disregards: monetise(-disregard),
-    }
-
-    transaction_allowance = main_home.fetch(:transaction_allowance)
-    data[:deductions] = monetise(-transaction_allowance) if transaction_allowance.positive?
-
-    data
+  def additional_property_data(prefix:)
+    additional_property = capital_items(:properties, prefix)[:additional_properties].first
+    property_data(additional_property, property_type: :additional)
   end
 
-  def additional_property_rows(prefix:)
-    home = capital_items(:properties, prefix)[:additional_properties].first
+  def property_data(property, property_type:)
     data = {
-      value: monetise(home.fetch(:value)),
-      mortgage: monetise(-home.fetch(:outstanding_mortgage)),
+      value: property.fetch(:value),
+      mortgage: -property.fetch(:outstanding_mortgage),
+      transaction_allowance: -property.fetch(:transaction_allowance),
```

Comment:
general question about `transaction_allowance` - how is this calculated? When is it sent to us by CFE?

Reply:
Ok, so broadly what happens is we collect information about the client's finances on each successive form screen, and then, when the user hits 'Submit' on the "Check your answers" screen, we send all the information over to CFE in one go (well, not quite in one go - the CFE API is designed so that we have to call a series of endpoints one by one to pass the data we've collected to it), and CFE then crunches all the numbers and passes back to us a big payload containing the overall result ("eligible"/"ineligible"/"contribution_required") as well as a breakdown of all of the workings. As part of the property calculation, in specific circumstances it applies a 3% transaction allowance to represent the cost of sale - if your home is worth £250k and you sold it, you wouldn't end up with £250k in your bank account thanks to lawyers fees etc. So CFE's "property" payload contains everything to say e.g "You said your home was worth £250k, we've subtracted 3% of that, and we've also subtracted a £100k disregard because it's your main home, so for the purposes of deciding how much capital you have we've decided your property is the equivalent of £142,500 in cash"